calico_containers/calicoctl.py

```diff
@@ -437,6 +461,64 @@ def normalize_version(version):
     """
     return [int(x) for x in re.sub(r'(\.0+)*$','', version).split(".")]
 
+
+def get_host_ips():
+    """
+    Searches for the first IP address (that isn't the docker0 veth).
+    Since already has a dependency on 'ip' binary, we'll use that to
+    find the address.
+    """
+    ip = sh.Command._create("ip")
+    route_output = ip("route").stdout
+    # Search for "src IPADDR" and extract the IP for any interface
+    pattern = "src ([\d]{1,3}\.[\d]{1,3}\.[\d]{1,3}\.[\d]{1,3})"
+    ips = re.findall(pattern, route_output, re.MULTILINE)
+    return ips
+
+def get_host_ip6s():
+    """
+    Searches for the first IP address (that isn't the docker0 veth).
+    """
+    # Use the IP command since calico-docker has it as a dependency
+    ip = sh.Command._create("ip")
+    route_output = ip("-6", "route").stdout.strip()
+    routes = route_output.split("\n")
+    ip_6_addrs = []
+    for route in routes:
+        if "docker0" not in route:
+            ip_6_addr = route.split()[0]
```

Comment:
this is wrong.  The `ip -6 route` command doesn't give the local address assigned to the interface, just the prefix.  Try `ip -o -6 addr`